homeassistant/components/media_player/__init__.py

```diff
@@ -301,6 +307,26 @@ def media_seek_service(service):
     hass.services.register(DOMAIN, SERVICE_MEDIA_SEEK, media_seek_service,
                            descriptions.get(SERVICE_MEDIA_SEEK))
 
+    def select_source_service(service):
```

Comment:
You will also need to add a method to the root that calls this service. Like the other services. This new method also needs to be tested.